packages/pangea-node-sdk/src/services/authn.ts

```diff
@@ -0,0 +1,421 @@
+   * @param {String} o.email - An email address
+   * @param {Boolean | null} o.require_mfa - New require_mfa value
+   * @param {String | null} o.mfa_value - New MFA provider setting
+   * @param {String | null} o.mfa_provider - New MFA provider setting
+   * @param {Object} o.profile - Updates to a user profile
+   * @returns {Promise<PangeaResponse<AuthN.UserProfile>>} - A promise representing an async call to the endpoint
+   * @example
+   * const response = await authn.userProfileUpdate({
+   *   email: "joe.user@email.com",
+   *   profile: {
+   *     "phone": "18085967873",
+   *     "deleteme": null,
+   *   },
+   * });
+   */
+  userProfileUpdate(o: AuthN.UserProfileUpdateRequest): Promise<PangeaResponse<AuthN.UserProfile>> {
+    const valid = schema.object().shape({
+      identity: schema.optional().string(o.identity) || schema.optional().null(o.identity),
+      email: schema.optional().string(o.email) || schema.optional().null(o.email),
+      require_mfa:
+        schema.optional().boolean(o.require_mfa) || schema.optional().null(o.require_mfa),
+      mfa_value: schema.optional().string(o.mfa_value) || schema.optional().null(o.mfa_value),
+      mfa_provider:
+        schema.optional().string(o.mfa_provider) || schema.optional().null(o.mfa_provider),
+      // No need to run validation on o.profile
+      // since it's an object with a shape of any key value pair
+    });
+
+    if (!valid.identity) {
+      throw "userProfileUpdate was called without supplying a valid identity";
+    }
+    if (!valid.email) {
+      throw "userProfileUpdate was called without supplying a valid email";
+    }
+    if (!valid.require_mfa) {
+      throw "userProfileUpdate was called without supplying a valid require_mfa";
+    }
+    if (!valid.mfa_value) {
+      throw "userProfileUpdate was called without supplying a valid mfa_value";
+    }
+    if (!valid.mfa_provider) {
+      throw "userProfileUpdate was called without supplying a valid mfa_provider";
+    }
+
+    return this.post("user/profile/update", o);
+  }
+
+  // authn::/v1/user/update
+  /**
+   * @summary Administration user update
+   * @description Update user's settings
+   * @param {String | null} o.identity - The identity of a user or a service
+   * @param {String | null} o.email - An email address
+   * @param {String | null} o.authenticator - New value for a user's authenticator
+   * @param {Boolean | null} o.disabled - New disabled value. Disabling a user account will prevent them from logging in.
+   * @param {Boolean | null} o.require_mfa - New require_mfa value
+   * @returns {Promise<PangeaResponse<AuthN.UserUpdateResult>>} - A promise representing an async call to the endpoint
+   * @example
+   * const response = await authn.userUpdate({
+   *   email: "joe.user@email.com",
+   *   require_mfa: true,
+   * });
+   */
+  userUpdate(o: AuthN.UserUpdateRequest): Promise<PangeaResponse<AuthN.UserUpdateResult>> {
+    const valid = schema.object().shape({
```

Comment:
Is there a reason we don't just use yup for validation if we think it is needed?

Are we doing some param validation in the sdk to help save the user on some API calls? It is just going to be an extra thing to keep in sync is my only worry

Reply:
Pulled validation out of this pull request: https://github.com/pangeacyber/pangea-javascript/pull/63#discussion_r1103469756